src/webgpu/api/operation/resource_init/depth_stencil_attachment_clear.spec.ts

```diff
@@ -0,0 +1,242 @@
+export const description =
+  'Test uninitialized textures are initialized to zero when used as a depth/stencil attachment.';
+
+import * as C from '../../../../common/constants.js';
+import { TestGroup } from '../../../../common/framework/test_group.js';
+import { unreachable } from '../../../../common/framework/util/util.js';
+import {
+  fillTextureDataWithTexelValue,
+  getTextureCopyLayout,
+} from '../../../util/texture/layout.js';
+import { SubresourceRange } from '../../../util/texture/subresource.js';
+import { getTexelDataRepresentation } from '../../../util/texture/texelData.js';
+
+import {
+  InitializedState,
+  ReadMethod,
+  TextureZeroInitTest,
+  initializedStateAsDepth,
+  initializedStateAsStencil,
+} from './texture_zero_init_test.js';
+
+class DepthStencilAttachmentClearTest extends TextureZeroInitTest {
+  private depthTestPipelineCache: Map<string, GPURenderPipeline> = new Map();
+  private getDepthTestReadbackPipeline(
+    state: InitializedState,
+    format: GPUTextureFormat,
+    sampleCount: number
+  ): GPURenderPipeline {
+    const key = [state, format, sampleCount].join('_');
+    if (this.depthTestPipelineCache.has(key)) {
+      return this.depthTestPipelineCache.get(key)!;
+    }
+
+    const pipeline = this.device.createRenderPipeline({
+      vertexStage: {
+        entryPoint: 'main',
+        module: this.makeShaderModule('vertex', {
+          glsl: `#version 310 es
+          void main() {
+            const vec2 pos[3] = vec2[3](
+                vec2(-1.f, -3.f), vec2(3.f, 1.f), vec2(-1.f, 1.f));
+            gl_Position = vec4(pos[gl_VertexIndex], 0.f, 1.f);
+          }
+          `,
+        }),
+      },
+      fragmentStage: {
+        entryPoint: 'main',
+        module: this.makeShaderModule('fragment', {
+          glsl: `#version 310 es
+          precision highp float;
+          layout(location = 0) out float outSuccess;
+
+          void main() {
+            gl_FragDepth = float(${initializedStateAsDepth(state)});
+            outSuccess = 1.0;
+          }
+          `,
+        }),
+      },
+      colorStates: [
+        {
+          format: 'r8unorm',
+        },
+      ],
+      depthStencilState: {
+        format,
+        depthCompare: 'equal',
```

Comment:
nit: comment to briefly explain how the test works, here and in main() above; same for stencil

Reply:
Done